File: pset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int get_cents();

int main(void)
{
    int change = get_cents();
    int coins  = 0;
    int coin_type[] = {25, 10, 5, 1};

    for (int i = 0; i < 4; i++)
    {
        while ( change >= coin_type[i] )
        {
            coins++;
            change-= coin_type[i];
        }
    }
    printf("%i\n",coins);
}

int get_cents()
{
    float f;
    do
    {
        printf("O hai! How much change is owed?\n");
        f = get_float();
    }
    while (f < 0);
    return ((int) round(f*100) );
}

File: pset6/greedy.py
##
# Calculates the smallest amount of coins for a given change
##

import cs50


def main():
    change = get_cents()
    coins = 0
    coin_type = [25, 10, 5, 1]
    for i in range(4):
        while change >= coin_type[i]:
            coins += 1
            change -= coin_type[i]
    print(coins)
    exit(0)


def get_cents():
    while True:
        print("O hai! How much change is owed?")
        f = cs50.get_float()
        if f >= 0:
            break
    return round(f*100)

if __name__ == "__main__":
    main()
